src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -145,6 +130,35 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                     }
                     process = ProcessLauncher.Launcher.ChildProc;
                 }
+                else if (port.Length != 0)
+                {
+                    ReversedDiagnosticsServer server = new ReversedDiagnosticsServer(port);
+                    server.Start();
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        Console.WriteLine($"Waiting for connection on {port}");
+                        Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={port},suspend");
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Waiting for connection on {Path.GetFullPath(port)}");
+                        Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={Path.GetFullPath(port)},suspend");
+                    }
+
+                    ManualResetEvent acceptEvent = new ManualResetEvent(false);
```

Comment:
This entire block from the `ManualResetEvent acceptEvent = ...` to `acceptEvent.WaitOne()` can be replaced with
`IpcEndpointInfo endpointInfo = await server.AcceptAsync(ct);`. Otherwise, I feel like its jumping through a lot of hoops to mix async and non-async together for some reason.